### PR TITLE
[EOSF-885] Review action rename

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

[OSF-8876](https://openscience.atlassian.net/browse/OSF-8876) modifies the action API endpoint. This PR renames the Ember model from `action` to `review-action` and add tests for the changed model. There is [another PR](https://github.com/CenterForOpenScience/ember-osf-preprints/pull/498) for `ember-osf-preprints` repo for some minor changes.

## Summary of Changes

1. A bunch of renaming for files and imports.
2. In the newly renamed `adapters/review-action.js`, the default behavior of the adapter is overridden with a `pathForType` function, since the API endpoint for review action inquiry is changed from `/v2/actions/` to `/v2/actions/reviews`.
3. A bunch of tests for the newly renamed `review-action` model.

## Side Effects / Testing Notes

None

## Ticket

https://openscience.atlassian.net/browse/EOSF-885

# Reviewer Checklist

- [x] meets requirements
- [x] easy to understand
- [x] DRY
- [x] testable and includes test(s)
- [x] changes described in `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Moved share button in `file-browser-item` to the `file-browser` toolbar
 - Rename button to have class `primary` instead of `success` on the `file-browser` component
 - What screen sizes columns are displayed in `file-browser` table
+- Rename `action` model to `review-action`
+  - Also rename related adapter, serializer and tests
+  - Added model tests for `review-action`
+  - Customize model adapter
```
Comment (jamescdavis) [inline CHANGELOG.md]: Perfect. Thank you!

**Comment by jamescdavis** [comment]
@laurenbarker Oops, I keep forgetting there is a user relationship! Sorry about that!

**Comment by adlius** [comment]
A PR is coming up shortly.

**Comment by laurenbarker** [comment]
Thank you!

**Comment by adlius** [comment]
Rename completed. Since I touched the `user` model, I also took the liberty to add some tests.

File: tests/unit/models/user-test.js [hunk 2]
```diff
@@ -1,15 +1,116 @@
...
+});
+
+test('review-actions relationship', function(assert) {
+    var model = this.store().modelFor('user');
+    var relationship = Ember.get(model, 'relationshipsByName').get('reviewActions');
+
+    assert.equal(relationship.key, 'reviewActions');
+    assert.equal(relationship.type, 'review-action');
+    assert.equal(relationship.kind, 'hasMany');
+});
```
Comment (jamescdavis) [inline tests/unit/models/user-test.js]: Thank you for adding all these!

**Comment by laurenbarker** [comment]
@adlius @jamescdavis, sorry for the flip flop. It looks like the user's review-actions relationship doesn't exist anymore so it can be removed from the model.

**Comment by adlius** [comment]
@laurenbarker PR is updated to remove the review-action relationship for user and its respective model test.

**Comment by jamescdavis** [comment]
Passing build: https://travis-ci.org/adlius/ember-osf/builds/319232973

**Comment by jamescdavis** [comment]
Still passing: https://travis-ci.org/adlius/ember-osf/builds/319331940

**Pull request merged**